packages/core/src/lib/light_push/light_push.ts

```diff
@@ -79,10 +141,35 @@ export class LightPushCore {
     encoder: IEncoder,
     message: IMessage,
     peerId: PeerId
-  ): Promise<CoreProtocolResult> {
+  ): Promise<LightPushCoreResult> {
+    // Get protocol and stream first to know which message format to use
+    let stream: Stream;
+    let protocol: string;
+    try {
+      const result = await this.getProtocolStream(peerId);
+      stream = result.stream;
+      protocol = result.protocol;
+    } catch (error) {
+      // Fallback to v2 if protocol detection fails
```

Comment:
You are already doing a fallback in https://github.com/waku-org/js-waku/pull/2501/files#r2239307113 and even so, I don't think you should fallback to v2 if v3 fails.
Only fallback to v2 if v3 is not supported by the remote peer.